src/solvers/dgmulti/shock_capturing.jl

```diff
@@ -0,0 +1,284 @@
+      neighboring_element = cache.EToE[face, element]
+      alpha_neighbor = alpha_tmp[neighboring_element]
+      alpha[element]  = max(alpha[element], 0.5 * alpha_neighbor)
+    end
+  end
+
+end
+
+#     pure_and_blended_element_ids!(element_ids_dg, element_ids_dgfv, alpha, dg, cache)
+#
+# Given blending factors `alpha` and the solver `dg`, fill
+# `element_ids_dg` with the IDs of elements using a pure DG scheme and
+# `element_ids_dgfv` with the IDs of elements using a blended DG-FV scheme.
+function pure_and_blended_element_ids!(element_ids_dg, element_ids_dgfv, alpha,
+                                       mesh::DGMultiMesh, dg::DGMulti)
+  empty!(element_ids_dg)
+  empty!(element_ids_dgfv)
+
+  for element in eachelement(mesh, dg)
+    # Clip blending factor for values close to zero (-> pure DG)
+    dg_only = isapprox(alpha[element], 0, atol=1e-12)
+    if dg_only
+      push!(element_ids_dg, element)
+    else
+      push!(element_ids_dgfv, element)
+    end
+  end
+
+  return nothing
+end
+
+function calc_volume_integral!(du, u,
+                               mesh::DGMultiMesh,
+                               have_nonconservative_terms, equations,
+                               volume_integral::VolumeIntegralShockCapturingHG,
+                               dg::DGMultiFluxDiff, cache)
+
+  (; element_ids_dg, element_ids_dgfv) = cache
+  (; volume_flux_dg, volume_flux_fv, indicator) = volume_integral
+
+  # Calculate blending factors α: u = u_DG * (1 - α) + u_FV * α
+  alpha = @trixi_timeit timer() "blending factors" indicator(u, mesh, equations, dg, cache)
+
+  # Determine element ids for DG-only and blended DG-FV volume integral
+  pure_and_blended_element_ids!(element_ids_dg, element_ids_dgfv, alpha, mesh, dg)
+
+  # Loop over pure DG elements
+  @trixi_timeit timer() "pure DG" @threaded for idx_element in eachindex(element_ids_dg)
+    element = element_ids_dg[idx_element]
+    flux_differencing_kernel!(du, u, element, mesh, have_nonconservative_terms,
+                              equations, volume_flux_dg, dg, cache)
+  end
+
+  # Loop over blended DG-FV elements, blend the high and low order RHS contributions
+  # via `rhs_high * (1 - alpha) + rhs_low * (alpha)`.
+  @trixi_timeit timer() "blended DG-FV" @threaded for idx_element in eachindex(element_ids_dgfv)
+    element = element_ids_dgfv[idx_element]
+    alpha_element = alpha[element]
+
+    # Calculate DG volume integral contribution
+    flux_differencing_kernel!(du, u, element, mesh,
+                              have_nonconservative_terms, equations,
+                              volume_flux_dg, dg, cache, 1 - alpha_element)
+
+    # Calculate "FV" low order volume integral contribution
+    low_order_flux_differencing_kernel!(du, u, element, mesh,
+                                        have_nonconservative_terms, equations,
+                                        volume_flux_fv, dg, cache, alpha_element)
```

Comment:
We have called this `fv_kernel!` for the other solvers, I think. Maybe use consistent names? If you have a good reason to use the name `low_order_flux_differencing_kernel!`, I'm of course open to it (and would like to know the reason).

Reply:
I am ok with fv_kernel! too. I chose the other name since the fv interpretation doesn't hold on more general elements. 

Up to you on the name.

Reply:
Fair enough. Let's keep the name as it is.